_posts/2024-01-16-owasp-top10-with-abstract-examples-part-2.md

```diff
@@ -0,0 +1,120 @@
+tags:
+ - DevSecOps
+ - InfoSec
+ - OWASP
+ - Quality
+ - Security
+ - Software security
+ - Web development
+---
+
+## What is this blog about?
+
+This is the second part of our [OWASP top 10](https://owasp.org/Top10/) explanation with non-technical examples. In [part 1](https://dev.solita.fi/2024/01/09/owasp-top10-with-abstract-examples-part-1.html) I lead into the subject why understanding the most common risks for non-technical persons is also important and explained risks from one to three. Let's continue with risks four to ten.
+
+## 4 Insecure Design
+
+Applications can be designed with care and safety in mind from the beginning or with the fastest possible return of investment in mind. Unfortunately, the latter mostly means that information security gives way to "more important features". [Insecure design](https://owasp.org/Top10/A04_2021-Insecure_Design/) means that security is not built in, but is added afterwards when the product is ready enough. This fails in most cases and is more expensive. Since it is more expensive to do, it is either hardly done or not done at all. For example, application layers or the network can be taken into account in the design. Fixing these later is expensive.
+
+![Insecure design](/img/2024-01-owasp-top-10/4_insecure_design_3.jpg)
+
+What if in our scenario the business owners were busy building a restaurant building. The sales window could be implemented so that the computer is at such an angle that the customer can see the screen along with other sensitive information. Another good example would be that the hatch locking mechanism is too complicated to use. As a result, employees just keep it open because they are lazy.
+
+### The fix
+
+Try to enable Secure Software Development Lifecycle. It means that security has moved to the left in the software development life cycle. When starting a project, think about the risks considering the project and the product. What laws and conditions must be taken into account. Test and raise issues on time, etc. Read more about SSDLC in [Joona's thorough blog post](https://dev.solita.fi/2020/04/08/secure-software-development-lifecycle.html).
+
+In our example, business owners should know how to run a business in this kind of restaurant. They should have designed the facilities before there was brick on top of brick. Otherwise, the drive-thru does not work as a drive-thru and business stalls.
+
+## 5 Security Misconfiguration
+
+Most of the [security misconfigurations](https://owasp.org/Top10/A05_2021-Security_Misconfiguration/) are done by accident, but deliberately weakening protections, such as closing a firewall or leaving unnecessary network ports open, are also done on purpose. Forgetting to close unnecessary features or apis, leaking dangerous information in error messages, using default credentials and not segmenting the network are also good examples of security mistakes.
```

Comment:
"apis" => "APIs" 
https://english.stackexchange.com/questions/503/what-is-the-correct-way-to-pluralize-an-acronym-initialism